src/hardware/hantek-6xxx/api.c

```diff
@@ -707,6 +719,7 @@ static void LIBUSB_CALL receive_transfer(struct libusb_transfer *transfer)
 		devc->sample_buf = NULL;
 		sdi->driver->dev_acquisition_stop(sdi, NULL);
 	} else {
+		libusb_free_transfer(transfer);
```

Comment:
everytime we call read_channel, this calls hantek_6xxx_get_channeldata which does a 
transfer = libusb_alloc_transfer(0);
but the corresponding free call is missing IMHO. So I tried to add this to see if the problems with samples sizes > 10M would go away if I free the transfer structure. It did not, but I still think we might have a memory leak if don't do this... But I did not test with valgrind or similar...